pkg/cloudprovider/util/cloud_init_settings.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2021 The Machine Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"html/template"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	CloudInitNamespace = "cloud-init-settings"
```

Comment:
Nit: the namespace may be too general. Maybe we should call it `machine-controller` or `machine-controller-cloud-init`.

Reply:
hmm I am not sure, as this is a user cluster and there is no other place where we get the cloud init from. So not quite sure if this is really general :)